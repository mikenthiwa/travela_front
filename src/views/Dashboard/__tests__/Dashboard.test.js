import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import ConnectedDashboard, { Dashboard, mapStateToProps } from '..';
import checkUserPermission from '../../../helper/permissions';


const initialState = {
  analytics: {
    payload: {
      total_requests: 230
    },
    error: '',
    isLoading: false,
    success: false
  }
};

const props = {
  getCurrentUserRole: ['Travel Administrator'],
  history: {
    push: jest.fn()
  },
  departmentTrips: {
    report: [],
    loading: false,
  },
  fetchDepartmentTrips: jest.fn(),
  fetchReadiness: jest.fn(),
  exportReadiness: jest.fn(),
  downloadCsv: jest.fn(),
  downloadAnalytics: jest.fn(),
  downloadCalendarAnalytics: jest.fn(),
  fetchCalendarAnalytics: jest.fn(),
  travelCalendar: {
    isLoading: false,
    travelCalendarData: {
      data:[],
      pagination:{},
      travelCalendarError: 'No records found',
    }
  },
  readiness: {
    isLoading: false,
  },
  isLoaded: false,
  currentUser: {
    id: 2,
    email: 'william.sserubiri@andela.com'
  }
};

localStorage.setItem('location', 'Nairobi, Kenya');
jest.mock('../../../helper/permissions', () => jest.fn());

describe('<Dashboard />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Dashboard {...props} />);
  });


  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('maps the correct state to props', () => {
    const mapper = mapStateToProps(
      {
        user : {getCurrentUserRole: ['Travel Admin']},
        analytics: {departmentTrips: {
          report: [],
          loading: false,
        }}
      });
    expect(mapper.getCurrentUserRole[0]).toEqual('Travel Admin');
  });
});
