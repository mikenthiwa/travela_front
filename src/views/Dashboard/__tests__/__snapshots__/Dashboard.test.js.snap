// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<div
  id="dashboard"
>
  <FilterContext
    history={
      Object {
        "push": [Function],
      }
    }
    user={
      Object {
        "email": "william.sserubiri@andela.com",
        "id": 2,
      }
    }
  >
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </FilterContext>
  <TravelCalendar
    downloadCalendarAnalytics={[Function]}
    fetchCalendarAnalytics={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    travelCalendar={
      Object {
        "isLoading": false,
        "travelCalendarData": Object {
          "data": Array [],
          "pagination": Object {},
          "travelCalendarError": "No records found",
        },
      }
    }
    user={
      Object {
        "email": "william.sserubiri@andela.com",
        "id": 2,
      }
    }
  />
</div>
`;
